pkg/grpc/server.go

```diff
@@ -61,3 +77,83 @@ func (s *server) StartNonBlocking() error {
 	}()
 	return nil
 }
+
+func (s *server) stop() {
+	s.srv.GracefulStop()
+}
+
+func (s *server) getGRPCServer() (*grpc_go.Server, error) {
+	opts := []grpc_go.ServerOption{}
+
+	if s.tracingSpec.Enabled {
+		opts = append(opts, grpc_go.StreamInterceptor(diag.TracingGRPCMiddleware(s.tracingSpec)), grpc_go.UnaryInterceptor(diag.TracingGRPCMiddlewareUnary(s.tracingSpec)))
+	}
+
+	if s.authenticator != nil {
+		log.Info("sending workload csr request to sentry")
+		signedCert, err := s.authenticator.CreateSignedWorkloadCert(s.config.DaprID)
+		if err != nil {
+			return nil, fmt.Errorf("error from authenticator CreateSignedWorkloadCert: %s", err)
+		}
+
+		tlsCert, err := tls.X509KeyPair(signedCert.WorkloadCert, signedCert.PrivateKeyPem)
+		if err != nil {
+			return nil, fmt.Errorf("error creating x509 Key Pair: %s", err)
+		}
+
+		caCertPool := signedCert.TrustChain
+		tlsConfig := tls.Config{
+			Certificates: []tls.Certificate{tlsCert},
+			ClientCAs:    caCertPool,
+			ClientAuth:   tls.VerifyClientCertIfGiven,
+		}
+		ta := credentials.NewTLS(&tlsConfig)
+
+		log.Info("certificate signed successfully")
+
+		opts = append(opts, grpc_go.Creds(ta))
+		go s.startWorkloadCertRotation(signedCert.Expiry)
+	}
+	return grpc_go.NewServer(opts...), nil
+}
+
+func (s *server) startWorkloadCertRotation(expiry time.Time) {
```

Comment:
@yaron2 Can we use [reload feature](https://github.com/grpc/grpc/pull/12644/files) instead of restarting ? restarting server would make downtime. reloading cert would be cheaper and seemless

cc/ @amanbha @artursouza